app/Policies/UserPolicy.php

```diff
@@ -122,19 +125,36 @@ public function updateAnyPermission(User $user, User $target, Role $role = null)
         if (!isset($role)) {
             return $user->hasRole(Role::SECRETARY)
             || $user->isInStudentsCouncil()
+            || $user->isStudentsCouncilSecretary()
             || $user->hasAnyRoleBase([Role::WORKSHOP_ADMINISTRATOR, Role::WORKSHOP_LEADER]);
         }
 
         if ($role->name == Role::COLLEGIST) {
             return $user->hasRole(Role::SECRETARY);
         }
 
+        if ($role->name == Role::WORKSHOP_LEADER) {
+            return $user->hasRole(Role::SECRETARY) || $user->hasRole(Role::DIRECTOR);
+        }
+
+        if ($role->name == Role::WORKSHOP_ADMINISTRATOR) {
+            return $user->hasRoleBase(Role::WORKSHOP_LEADER) || $user->hasRole(Role::STUDENT_COUNCIL_SECRETARY) || $user->hasRole(Role::SECRETARY) || $user->hasRole(Role::STUDENT_COUNCIL, Role::SCIENCE_VICE_PRESIDENT);
+        }
+
+        if ($role->name == Role::STUDENT_COUNCIL_SECRETARY) {
+            return $user->isStudentsCouncilSecretary();
+        }
+
+        if ($role->name == Role::BOARD_OF_TRUSTEES_MEMBER) {
+            return $user->isStudentsCouncilSecretary();
```

Comment:
Don't we want to add the president here?

Reply:
I don't think that they should be able to assign the board of trustees members.